### PR TITLE
Defect/173 report coverage date overlap

### PR DESCRIPTION
**Pull request by Elaine-Krauss-TCG**
API Ticket [173](https://app.zenhub.com/workspaces/fecfile-online-619e578e68408b001c831251/issues/fecgov/fecfile-web-api/173)

Updates the coverage date overlap checking to catch an edge case where a new report is created whose dates entirely encompass another report's dates.  Adds unit tests to verify that the edge case has been caught.

### CODE REVIEW

**Comment by Elaine-Krauss-TCG** [comment]
Shelly has pointed out that a comment on the ticket adds fixing the coverage date fields to the requirements.  A fix for that has been added.

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts [hunk 1]
```diff
@@ -101,29 +101,68 @@ export class CreateF3XStep1Component implements OnInit, OnDestroy {
...
+      this.buildCoverageDatesValidator('coverage_through_date'),
+      Validators.required
+    ]);
 
     // Initialize validation tracking of current JSON schema and form data
     this.validateService.formValidatorSchema = f3xSchema;
     this.validateService.formValidatorForm = this.form;
+
+    //Re-checks the validators on the coverage date fields every second
+    setInterval(()=>{
```
Comment (toddlees) [inline front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts]: I think we really want to avoid running validation on a time schedule.  What prompted this addition?

**Review by toddlees — DISMISSED** [review]

File: front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts (hunk 1 above)
Comment (Elaine-Krauss-TCG) [reply front-end/src/app/reports/f3x/create-workflow/create-f3x-step1.component.ts]: Without this, the validation for a field only runs on form submission and on interaction with the given field.  Normally, this is just fine, but with the edge case this ticket is trying to solve, both fields need to show the validation error because neither one field is at fault.  This is something I would avoid normally, but it seems necessary here, and it comes at a negligible performance cost.  (I ran a performance profile on my machine using the dev panel, and the total runtime of timer events in a 30 second window came out to 180ms)

**Review by mjtravers — APPROVED** [review]
